ci/test_cpp_common.sh

```diff
@@ -5,6 +5,8 @@ set -euo pipefail
 
 . /opt/conda/etc/profile.d/conda.sh
 
+RAPIDS_VERSION_MAJOR_MINOR="$(rapids-version-major-minor)"
```

Comment:
In `build_docs.sh` we `export` this, but here we're just setting it locally. I doubt it matters here but it might be worth establishing consistent patterns.

Reply:
My thinking here was "export it if it's already exported (as in the docs example), otherwise make it local", since I thought that had the least risk of causing other surprising problems.

I clicked around a little, and I think:

* `RAPIDS_VERSION` = does not even need to be set in docs-build scripts
* `RAPIDS_VERSION_MAJOR_MINOR` = does not need to be exported
* `RAPIDS_VERSION_NUMBER` = needs to be exported, for `rapids-upload-docs` ([code link in gha-tools](https://github.com/rapidsai/gha-tools/blob/c5a6fef8b6455d524aefe3baf2432a68872e5b8f/tools/rapids-upload-docs#L36))

For all the other scripts that are not docs builds, this PR only introduces NEW variables, so I think not exporting them is slightly safer.

Just pushed https://github.com/rapidsai/cudf/pull/17013/commits/16367a5ae7967856347c7c433bb6b95dbf5b21f8 doing that. If that works, I'll make similar changes in the other PRs.

Reply:
This went fine in CI... docs built successfully and I see the expected versions of things in the environment. Docs look ok at the preview too: https://downloads.rapids.ai/ci/cudf/pull-request/17013/16367a5/docs/libcudf/html/index.html

Reply:
Some of these claims I made about the variables not needing to be exported were incorrect. Addressed in https://github.com/rapidsai/cudf/pull/17042 (and updated across all the other PRs for this issue).